### PR TITLE
[MI-3113] Fix email case sensitivity issue

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Fix the issue of the user not being able to connect due to an email case sensitivity issue

### CODE REVIEW

File: server/api.go [hunk 1]
```diff
@@ -350,7 +350,7 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if mmUser.Id != a.p.GetBotUserID() && msteamsUser.Mail != mmUser.Email {
```
Comment (jespino) [inline server/api.go:353]: This change shouldn't be needed because the client layer should be taking care of that entirely.

**Review by jespino — CHANGES_REQUESTED** [review]
A tiny change needed, but otherwise looks good to me.

**Comment by ayusht2810** [comment]
@jespino updated the PR. Please re-review

**Review by jespino — APPROVED** [review]
LGTM

**Pull request merged**